assignment2/assignment2.py

```diff
@@ -0,0 +1,418 @@
+        You convert the date strings into datetime objects using datetime.strptime(string, "%B %d, %Y").
+        So, you could use the following lambda: lambda x: (x[0], datetime.strptime(x[1], "%B %d, %Y"))
+        The function should return the resulting list.
+
+    Call the function from within your program.  Store the return value in the minutes_list global.  Print it out, so you can see what it looks like.
+
+    Run the test until the next part passes.
+    """
+    minutes_list = list(minutes_set)
+
+    minutes = map(
+        lambda row: (row[0], datetime.strptime(row[1], "%B %d, %Y")), minutes_list
+    )
+
+    return list(minutes)
+
+
+minutes_list = create_minutes_list()
+
+
+# Task 15
+def write_sorted_list():
+    """
+
+    Create a function called write_sorted_list.  It takes no parameters.  It should do the following:
+        Sort minutes_list in ascending order of datetime.
+        Call map again to convert the list. In this case, for each tuple, you create a new tuple. The first element of the tuple is the name (unchanged). The second element of the tuple is the datetime converted back to a string, using datetime.strftime(date, "%B %d, %Y")
+        Open a file called ./minutes.csv. Use a csv.writer to write out the resulting sorted data. The first row you write should be the value of fields the from minutes1 dict. The subsequent rows should be the elements from minutes_list.
+        The function should return the converted list.
+
+    Call this function from within your program.  Then check that the file is created, and that it contains appropriate content.
+
+    Run the test again until the next test has passed.
+    """
+    minutes_list.sort(key=lambda row: row[1])
+    minutes = list(
+        map(lambda row: (row[0], datetime.strftime(row[1], "%B %d, %Y")), minutes_list)
+    )
+
+    with open("./minutes.csv", "w") as file:
+        writer = csv.writer(file)
+
+        for row in minutes:
+            writer.writerow(row)
+
+    return minutes
+
+
+minutes = write_sorted_list()
+print(minutes)
+
+"""Check for Understanding
+
+    You created the minutes_set from several lists.  You then created a list from the set.  What's the point of the set, if you're going to end up with a list?
+
+    Why did you subsequently need to create the list called minutes_list?  Couldn't you just keep working with the set?
+
+    Why did you need to convert the date strings to datetime objects?
+
+    Why did you convert them back to strings before writing out the CSV?
+"""
+# the answers:
+# We used a set to get rid of duplicate values so we converted to sets
+# in order to use mutable lists we have to convert set back to list
+# the date string was converted to allow easier sorting via datetime
+# the datetime object needs to be converted to str to be printed in a prededined way
```

Comment:
Excellent work and good answers!!!

Week 2 is one of the toughest in the set, partly because it uses things that aren't seen much in Python anymore (lambdas) and are hard to wrap your head around. Good work!